### PR TITLE
Improve field alignment

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Use https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/fieldalignment to improve the alignment of fields in structs, which reduces memory usage, and reduces GC scan effort.

NOTE: Unfortunately, `fieldalignment` does not preserve comments when fixing code. So for cases where fields had comments, I manually inserted them back, so this is something to look out for.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by dreamsmasher — APPROVED** [review]
LGTM. i wish the compiler would handle this for you like rustc does

**Review by dsainati1 — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

File: runtime/ast/expression.go [hunk 1]
```diff
@@ -802,11 +802,9 @@ type AccessExpression interface {
 
 type MemberExpression struct {
 	Expression Expression
-	Optional   bool
-	// The position of the token (`.`, `?.`) that separates the accessed expression
-	// and the identifier of the member
-	AccessPos  Position
```
[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/ast/expression.go:808]: Yes, good catch!

File: runtime/ast/expression.go [hunk 2]
```diff
@@ -802,11 +802,9 @@ type AccessExpression interface {
...
 	Identifier Identifier
+	AccessPos  Position
```
[the remark this answers is not in the document]
Comment (dreamsmasher) [reply runtime/ast/expression.go:808]: this wouldn't have happened if the compiler handled alignment for you 🤔

[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/ast/expression.go:808]: @dreamsmasher Yes, you made your point. You already mentioned this above. Please avoid further unconstructive feedback like this

**Pull request merged**